packages/runtime/src/controls/style-v2.ts

```diff
@@ -0,0 +1,79 @@
+import { BoxModel } from 'css-box-model'
+import { ControlDefinition, ControlDefinitionData } from './control'
+import { PropController } from '../prop-controllers/base'
+import { Send } from '../prop-controllers/instances'
+import { ResponsiveValue } from './types'
+import { CSSObject } from '@emotion/serialize'
+import { useStyle } from '../runtimes/react/use-style'
+
+export const StyleV2ControlType = 'makeswift::controls::style-v2'
+
+export const StyleV2Format = {
+  Class: 'makeswift::controls::style-v2::format::class',
+  Object: 'makeswift::controls::style-v2::format::object',
+} as const
+
+export type StyleV2Format = typeof StyleV2Format[keyof typeof StyleV2Format]
+
+export type StyleV2CSSObject = CSSObject
+export const useUnstableStyleV2ClassName = useStyle
+
+type StyleV2ControlConfig<T extends ControlDefinition = ControlDefinition> = {
+  type: T
+  getStyle(item: ControlDefinitionData<T> | undefined): CSSObject
+  format: StyleV2Format
+}
+
+export type StyleV2ControlDefinition = {
+  type: typeof StyleV2ControlType
+  config: StyleV2ControlConfig
+}
+
+export function unstable_StyleV2<C extends StyleV2ControlConfig>(
+  config: C,
+): StyleV2ControlDefinition {
+  return { type: StyleV2ControlType, config }
+}
+
+/**
+ * TODO: Rewrite `ControlDefinitionData` as a tail recursive type.
+ *
+ * Explanation: Making `StyleV2ControlData` generic causes us to hit a limitation of TS conditional types for the definition of `ControlDefinitionData`.
+ * We get "Type instantiation is excessively deep and possibly infinite."
+ *
+ * Helpful links:
+ * * The PR changing the limit of conditional types: https://github.com/microsoft/TypeScript/pull/45711
+ * * TS 4.5 docs releasing this limitation: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-5.html#tail-recursion-elimination-on-conditional-types
+ * * A SO user implementing a tail recursive type: https://stackoverflow.com/questions/72370456/tail-recursion-elimination-on-conditional-types-doesnt-work
+ *
+ *
+ * This is what the type should be:
+ * ```
+ * export type StyleV2ControlData<T extends StyleV2ControlDefinition = StyleV2ControlDefinition> =
+ *   ResponsiveValue<ControlDefinitionData<T['config']['type']>>
+ * ```
+ */
+export type StyleV2ControlData = ResponsiveValue<any>
```

Comment:
Decided on any typing the control value here @migueloller. I think we are actually reaching a limitation of the conditional type for `ControlDefinitionData`

There might be a way to rewrite `ControlDefinitionData` so it is "tail recursive." I feel out of my league though so I am going to any type this for now.

Reply:
We might want to put the `any` where the limitation is happening, i.e., the inference of the data instead of there in the underlying data definition, which will be used in various places

Reply:
Opting to keep it here. The error happens on typings for List at _random_ locations that pass the threshold of what TS supports. I think it's better to have the any at one locations with the comment explaining it's presence.

Reply:
What I'm suggesting would be just at 1 location, `StyleV2ControlData<any>` in the conditional type `ControlData`. That way in this file when we implement functions such as `copy`, `merge`, etc. those can be properly typed

Reply:
I wish I had responded to this last week. I did try this and I remember getting the same error. There were a lot of roadblocks to this work, so I ended up just skipping this one and proceeding. 

When I ultimately work on this I plan to deal with this problem